monitor_api_services.go

```diff
@@ -0,0 +1,121 @@
+	Stats     EntityMetricsStats `json:"stats"`
+	Status    string             `json:"status"`
+}
+
+// EntityMetricsData describes the result and type for the entity metrics
+type EntityMetricsData struct {
+	Result     []EntityMetricsDataResult `json:"result"`
+	ResultType string                    `json:"result_type"`
+}
+
+// EntityMetricsDataResult contains the information of a metric and values
+type EntityMetricsDataResult struct {
+	Metric map[string]any `json:"metric"`
+	Values [][]any        `json:"values"`
+}
+
+// EntityMetricsStats shows statistics info of the metrics fetched
+type EntityMetricsStats struct {
+	ExecutionTimeMsec int    `json:"executionTimeMsec"`
+	SeriesFetched     string `json:"seriesFetched"`
+}
+
+// EntityMetricsFetchOptions are the options used to fetch metrics with the entity ids provided
+type EntityMetricsFetchOptions struct {
+	// EntityIDs are expected to be type "any" as different service_types have different variable type for their entity_ids. For example, Linode has "int" entity_ids whereas object storage has "string" as entity_ids.
+	EntityIDs []any `json:"entity_ids"`
+
+	Filters              []MetricFilter              `json:"filters,omitempty"`
+	Metrics              []EntityMetric              `json:"metrics"`
+	TimeGranularity      []MetricTimeGranularity     `json:"time_granularity,omitempty"`
+	RelativeTimeDuration *MetricRelativeTimeDuration `json:"relative_time_duration,omitempty"`
+	AbsoluteTimeDuration *MetricAbsoluteTimeDuration `json:"absolute_time_duration,omitempty"`
+}
+
+// MetricFilter describes individual objects that define dimension filters for the query.
+type MetricFilter struct {
+	DimensionLabel string               `json:"dimension_label"`
+	Operator       MetricFilterOperator `json:"operator"`
+	Value          string               `json:"value"`
+}
+
+// EntityMetric specifies a metric name and its corresponding aggregation function.
+type EntityMetric struct {
+	Name              string            `json:"name"`
+	AggregateFunction AggregateFunction `json:"aggregate_function"`
+}
+
+// MetricTimeGranularity allows for an optional time granularity setting for metric data.
+type MetricTimeGranularity struct {
+	Unit  MetricTimeUnit `json:"unit"`
+	Value int            `json:"value"`
+}
+
+// MetricRelativeTimeDuration specifies a relative time duration for data queries
+type MetricRelativeTimeDuration struct {
+	Unit  MetricTimeUnit `json:"unit"`
+	Value int            `json:"value"`
+}
+
+// MetricAbsoluteTimeDuration specifies an absolute time range for data queries
+type MetricAbsoluteTimeDuration struct {
+	Start time.Time `json:"start,omitempty"`
+	End   time.Time `json:"end,omitempty"`
+}
+
+// FetchEntityMetrics returns metrics information for the individual entities within a specific service type
+func (mc *MonitorClient) FetchEntityMetrics(ctx context.Context, serviceType string, opts *EntityMetricsFetchOptions) (*EntityMetrics, error) {
```

Comment:
```suggestion
// GetEntityMetrics returns metrics information for the individual entities within a specific service type
func (mc *MonitorClient) GetEntityMetrics(ctx context.Context, serviceType string, opts *GetEntityMetricsOptions) (*EntityMetrics, error) {
```

Should we change `Fetch` to `Get` to be consistent with the [API doc](https://techdocs.akamai.com/linode-api/reference/post-read-metric)?

Reply:
Using `fetch` here is because this endpoint is actually a `POST`. However, the prefix `Get` usually indicates an actual `GET` endpoint, which could be confusing. I've talked to the ACLP team about this situation, and they suggested `fetch` might be a better way to go. 

I'm not sure if using a `ServiceType` string is better here. Because the `ServiceType` enum includes a list of services, however, here we only accept dbaas for now. It could be misleading that we also support other service types.

Reply:
The `ServiceType` was defined for another set of endpoints, Dashboard. It's a separate effort though. From what I've learnt there isn't a clear plan for what services will be support in the next for Metrics here.